src/Sylius/Bundle/CoreBundle/EventListener/UserCartRecalculationListener.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\EventListener;
+
+use Sylius\Bundle\UserBundle\Event\UserEvent;
+use Sylius\Component\Cart\Provider\CartProviderInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\OrderProcessing\OrderRecalculatorInterface;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Symfony\Component\Security\Http\Event\InteractiveLoginEvent;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class UserCartRecalculationListener
+{
+    /**
+     * @var CartProviderInterface
+     */
+    private $cartProvider;
+
+    /**
+     * @var OrderRecalculatorInterface
+     */
+    private $orderRecalculator;
+
+    /**
+     * @param CartProviderInterface $cartProvider
+     * @param OrderRecalculatorInterface $orderRecalculator
+     */
+    public function __construct(CartProviderInterface $cartProvider, OrderRecalculatorInterface $orderRecalculator)
+    {
+        $this->cartProvider = $cartProvider;
+        $this->orderRecalculator = $orderRecalculator;
+    }
+
+    /**
+     * @param UserEvent $event
+     */
+    public function recalculateCartWhileLogin(UserEvent $event)
+    {
+        $this->recalculateCart();
+    }
+
+    /**
+     * @param InteractiveLoginEvent $event
+     */
+    public function recalculateCartWhileInteractiveLogin(InteractiveLoginEvent $event)
```

Comment:
Since we're not using `$event` at all we can typehint to `Symfony\Component\EventDispatcher\Event` class and have one method for both events ;)